backend/lib/kubernetes-client/mixins.js

```diff
@@ -0,0 +1,358 @@
+//
+// Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+'use strict'
+
+const { isPlainObject, isEmpty } = require('lodash')
+const { join } = require('path')
+const { Mixin } = require('mixwith')
+
+const WatchBuilder = require('./WatchBuilder')
+const { http } = require('./symbols')
+const { clusterScopedUrl, namespaceScopedUrl, setPatchType, PatchType } = require('./util')
+
+const V1Alpha1 = superclass => class extends superclass {
+  static get version () {
+    return 'v1alpha1'
+  }
+}
+
+const V1Beta1 = superclass => class extends superclass {
+  static get version () {
+    return 'v1beta1'
+  }
+}
+
+const V1 = superclass => class extends superclass {
+  static get version () {
+    return 'v1'
+  }
+}
+
+const CoreGroup = superclass => class extends superclass {
+  [http.prefixUrl] (url) {
+    return join(url, 'api', this.constructor.version)
+  }
+}
+
+const NamedGroup = superclass => class extends superclass {
+  [http.prefixUrl] (url) {
+    return join(url, 'apis', this.constructor.group, this.constructor.version)
+  }
+}
+
+const NamespaceScoped = Mixin(superclass => class extends superclass {
```

Comment:
refer to docu and add comment with reason for using mixin

Reply:
✔️

Reply:
added unit tests for mixins

Reply:
✔️